### ISSUE TITLE
[BUG] code using CSS pseudo elements not showing (particularly ::before and ::after)

### ISSUE THREAD

**Issue by LikeMoneySecretly** [microsoft/playwright#14390]
**Context:**
- Playwright Version: [latest as of writing]
- Operating System: [e.g. Windows, Linux or Mac] Windows 11
- Node.js version: [e.g. 12.22, 14.6]
- Browser:All
- Extra: Environment doesn't really matter,

Python Code Snippet:

```
from playwright.sync_api import sync_playwright
import time

with sync_playwright() as p:
    browser = p.chromium.launch(headless=False, slow_mo=10,channel="msedge")
    page = browser.new_page()
    page.goto("https://www.bet365.com/#/HO/")
    print("This is the page title:"+page.title())
    page.wait_for_timeout(100000)
    browser.close()
```

 **Gyazo Links to the problem website code below:**
[https://gyazo.com/25c80802004a43048dc9085144afa6ae](With Playwright opening edge)
[https://gyazo.com/05df109591e8ec8e29b6cc8fa530a608](Opened manually in edge)

Basically the issue is that code that relies on ::after is not showing up in the playwright version of the website

This is repeatable in chromium as well i just did it in edge for ease. My initial suspicion was that some of the anti botting software we use might be doing this, However this doesn't make sense as our typical response to botting is much stronger than simply making small parts of the website not show up. Therefore my impression was that it might actually be an issue with playwrights website interpretation, in particular that CSS dependant code is not getting interpreted as it should.

**Comment by LikeMoneySecretly** [comment]
https://gyazo.com/08271b7de2e054b9862be29b9736d532
I can go down further, it may be to do with the fact that the buttons and clickable containers that are meant to be showing are using multiple ::after s which may be influencing playwright perhaps?

**Comment by LikeMoneySecretly** [comment]
For exmaple https://gyazo.com/a21c7f0049794c858c15f26ec033113b

**Comment by LikeMoneySecretly** [comment]
I suppose the best way to describe it is that there are missing buttons and containers with variables that regularly change (see above for an example of one of those classes).

is it possible that playwright is only taking a Snapshot of a webpage and that it’s therefore failing to grab any information that changes relatively quickly after that initial snapshot. 

The reason I think that is that when opened in msedge with playwright it shows a blank area where those buttons and containers normally are.

**Comment by LikeMoneySecretly** [comment]
https://gyazo.com/8755706f37329e70fa46354a556fa9e1 This is probably the clearest example yet. The containers and buttons that don't appear have their class names crossed out in green. As you can see my suspicion is that playwright is treating the CSS differently than other code or that it's taking a weird version of the browser

**Comment by pavelfeldman** [comment]
There is no Playwright Edge, the same Microsoft Edge is running in both cases. We can't tell much by the DOM, can you observe the visual difference between actual pages? Could be different page, different point in time, etc. etc.

**Comment by LikeMoneySecretly** [comment]
Nah there is a clear visual difference, it’s also an issue that seems to happen in Puppeteer as well. When I ran the same test in puppeteer the identical issue happened. And no I copied the identical link between the two pages the playwright side is visually different and is clearly missing containers and buttons in the box area where they should be.

So to summarise, Same link, if both are refreshed from the same link at roughly the same time a difference appears (missing containers).

In regards to a point of time difference is there a way to record the internal playwright time/find out a difference between the way time works between the two browsers?

The time difference is really the only thing I can think that would carry between puppeteer and playwrighht

**Comment by LikeMoneySecretly** [comment]
I don’t suppose using page.wait_for_timeout(10000) is causing this issue? I use time.sleep(140) in the puppeteer version as well, just to enable me to keep the headless browser open long enough to check all the visual code through inspect element

**Comment by LikeMoneySecretly** [comment]
I tried using a system which waited for the specified Selectors to appear in the DOM before the browser quit however that did not work either. I think it might be an issue with the way playwright is handling waiting (something that is carried over from puppeteer maybe)

**Comment by LikeMoneySecretly** [comment]
I managed to fix it by using Puppeteer and connecting to an existing browser tab, does that help somewhat in figuring out whats going on here?

**Comment by pavelfeldman** [comment]
Is `https://www.bet365.com/#/HO/` the page? I just tried loading it and it looks it loads and then reloads into a different content. It only does it for the first time it is opened in a fresh browser / profile.

When Playwright is testing page, it opens them as if they are opened in the fresh browser. That's why you can see a difference between opening the page in the browser where it was already opened and in the Playwright test. Try it out yourself - open incognito profile and load the page. See it load and then flash white and reload.

**Comment by LikeMoneySecretly** [comment]
Yeah that works, I noticed in puppeteer that it seems to auto resize when connecting to browser instances so I can see this being the issue.

**Comment by LikeMoneySecretly** [comment]
As an aside after this issue has been closed, do you think this will work in headless mode? As headless mode at least from a superficial glance doesn't seem to require a resolution framing
